misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java

```diff
@@ -137,7 +173,45 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
                             vals[1] = value;
                         }
                     });
-                    suite.addMockFunction(vals[0] + MOCK_ANNOTATION_DELIMITER + vals[1], functionName);
+
+                    if (bLangPackage == null) {
+                        setBlangPackage(attachmentNode);
+                    }
+
+                    if (vals[0].isEmpty()) {
+                        diagnosticLog.logDiagnostic(Diagnostic.Kind.ERROR, ((BLangFunction) functionNode).pos,
+                                "Module name cannot be empty");
+                    }
+
+                    if (vals[1].isEmpty()) {
+                        diagnosticLog.logDiagnostic(Diagnostic.Kind.ERROR, ((BLangFunction) functionNode).pos,
+                                "Function name cannot be empty");
+                    }
+
+                    SymbolTable symbolTable = SymbolTable.getInstance(compilerContext);
+                    Map<BPackageSymbol, SymbolEnv> packageEnvironmentMap = symbolTable.pkgEnvMap;
+                    PackageID functionToMockID = getPackageID(vals[0], packageEnvironmentMap);
+
+                    BType functionToMockType = getFunctionType(packageEnvironmentMap, functionToMockID, vals[1]);
+                    BType mockFunctionType = getFunctionType(packageEnvironmentMap, bLangPackage.packageID,
+                            ((BLangFunction) functionNode).name.toString());
+
+                    if (functionToMockType != null && mockFunctionType != null) {
+                        if (!typeChecker.isAssignable(mockFunctionType, functionToMockType)) {
+                            diagnosticLog.logDiagnostic(Diagnostic.Kind.ERROR, ((BLangFunction) functionNode).pos,
```

Comment:
This typecheck will be checking the entire signature, not just the param types. So it's not exactly right to complain about parameter mismatches. You can try using the standard incompatible types error message we have in the compiler for this.

Reply:
It's in the form: `incompatible types: expected 'T`' but found 'T2'`

Reply:
Changed to standard error